docs/en/stack/data-frames/dataframe-examples.asciidoc

```diff
@@ -0,0 +1,362 @@
+[role="xpack"]
+[testenv="basic"]
+[[dataframe-examples]]
+== {dataframe-cap} examples
+++++
+<titleabbrev>Examples</titleabbrev>
+++++
+
+beta[]
+
+This page provides examples of how to use {dataframe-transforms} to derive useful 
+insights from your data. All the examples use one of the 
+{kibana-ref}/add-sample-data.html[{kib} sample datasets]. For a more detailed, 
+step-by-step example, see 
+<<ecommerce-dataframes,Transforming your data with {dataframes}>>.
+
+* <<example-best-customers>>
+* <<example-airline>>
+* <<example-clientips>>
+
+[float]
+[[example-best-customers]]
+=== Finding your best customers
+
+In this example, we use the eCommerce orders sample dataset to find the customers 
+who spent the most in our hypothetical webshop. Let's transform the data such 
+that the destination index contains the number of orders, the total price of 
+the orders, the amount of unique products and the average price per order, 
+and the total amount of ordered products for each customer.
+
+[source,js]
+----------------------------------
+POST _data_frame/transforms/_preview
+{
+  "source": {
+    "index": "kibana_sample_data_ecommerce"
+  },
+  "dest" : {
+    "index" : "sample_ecommerce_orders_by_customer"
+  },
+  "pivot": {
+    "group_by": { <1>
+      "user": { "terms": { "field": "user" }}, 
+      "customer_id": { "terms": { "field": "customer_id" }}
+    },
+    "aggregations": {
+      "order_count": { "value_count": { "field": "order_id" }},
+      "total_order_amt": { "sum": { "field": "taxful_total_price" }},
+      "avg_amt_per_order": { "avg": { "field": "taxful_total_price" }},
+      "avg_unique_products_per_order": { "avg": { "field": "total_unique_products" }},
+      "total_unique_products": { "cardinality": { "field": "products.product_id" }}
+    }
+  }
+}
+----------------------------------
+// CONSOLE
+// TEST[skip:SETUP]
+
+<1> Two `group_by` fields have been selected. This means the {dataframe} will 
+contain a unique row per `user` and `customer_id` combination. Within this 
+dataset both these fields are unique. By including both in the {dataframe} it 
+gives more context to the final results.
+
+NOTE: In the example above, condensed JSON formatting has been used for easier 
+readability of the pivot object.
+
+The API returns the following response (note that this example contains the 
+response partially and shows only the first document object):
+
+[source,js]
+----------------------------------
+{
+  "preview" : [
+    {
+      "total_unique_products" : {
+        "avg" : 2.0
+      },
+      "customer_id" : "10",
+      "user" : "recip",
+      "order_id" : {
```

Comment:
I think this is the wrong output, since the preview API listed above doesn't include order_id or taxful_total_price, etc. Here's what I get when I run the API on 7.2:

> {
  "preview" : [
    {
      "total_order_amt" : 3946.9765625,
      "order_count" : 59.0,
      "total_unique_products" : 116.0,
      "avg_unique_products_per_order" : 2.0,
      "customer_id" : "10",
      "user" : "recip",
      "avg_amt_per_order" : 66.89790783898304
    },
    ... 
]
}

Reply:
P.S. I love that these examples use the Kibana sample data, which makes it super easy for me (and our users) to test!

Reply:
:o Huge thanks for catching this. I played with the API and eventually copied the wrong output.
Lesson learned!